### PR TITLE
add compute_instace extend system disk support

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_compute_instance.go [hunk 1]
```diff
@@ -961,6 +961,40 @@ func resourceComputeInstanceV2Update(d *schema.ResourceData, meta interface{}) e
...
+
+		if err != nil {
+			return fmt.Errorf("Error creating HuaweiCloud block storage client: %s", err)
+		}
+
+		systemDiskID := d.Get("system_disk_id").(string)
+
+		err = volumeactions.ExtendSize(blockStorageClient, systemDiskID, extendOpts).ExtractErr()
+		if err != nil {
+			return fmt.Errorf("Error extending huaweicloud_blockstorage_volume_v2 %s size: %s", systemDiskID, err)
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_compute_instance.go]: Error extending huaweicloud_compute_instance system disk size

File: huaweicloud/resource_huaweicloud_compute_instance.go [hunk 2]
```diff
@@ -961,6 +961,40 @@ func resourceComputeInstanceV2Update(d *schema.ResourceData, meta interface{}) e
...
+		}
+
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"extending"},
+			Target:     []string{"available", "in-use"},
+			Refresh:    VolumeV2StateRefreshFunc(blockStorageClient, systemDiskID),
+			Timeout:    d.Timeout(schema.TimeoutCreate),
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_compute_instance.go]: use TimeoutUpdate instead.

File: huaweicloud/resource_huaweicloud_compute_instance.go [hunk 3]
```diff
@@ -961,6 +961,40 @@ func resourceComputeInstanceV2Update(d *schema.ResourceData, meta interface{}) e
...
+			Delay:      10 * time.Second,
+			MinTimeout: 3 * time.Second,
+		}
+
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf(
+				"Error waiting for huaweicloud_blockstorage_volume_v2 %s to become ready: %s", systemDiskID, err)
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_compute_instance.go]: Error waiting for volume %s ...

File: website/docs/r/compute_instance.html.markdown [hunk 4]
```diff
@@ -229,7 +229,8 @@ The following arguments are supported:
 	* `co-p1`: high I/O(performance-optimized) disk type.
 	* `uh-l1`: ultra-high I/O(latency-optimized) disk type.
 
-* `system_disk_size` - (Optional) The system disk size in GB, The value range is 1 to 1024. Changing this creates a new server.
+* `system_disk_size` - (Optional) The system disk size in GB, The value range is 1 to 1024. Changing this parameter will update the disk. 
+    You can extend the disk by setting this parameter to a new value, which must be between current size and the max size. Shrinking the disk is not supported.
```
Comment (niuzhenguo) [inline website/docs/r/compute_instance.html.markdown]: need to be more specified for the `max size`, or just not mention it.

**Review by niuzhenguo — CHANGES_REQUESTED** [review]

File: huaweicloud/resource_huaweicloud_compute_instance.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_compute_instance.go]: done

File: huaweicloud/resource_huaweicloud_compute_instance.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_compute_instance.go]: done

File: website/docs/r/compute_instance.html.markdown (hunk 4 above)
Comment (Jason-Zhang9309) [reply website/docs/r/compute_instance.html.markdown]: done

**Pull request merged**